### PR TITLE
Create XYK Pool

### PR DESCRIPTION
**Pull request by jvonasek**

### CODE REVIEW

File: src/sections/pools/modals/CreateXYKPool/CreateXYKPool.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                assetA={assetA}
+                assetB={assetB}
+              />
+            ),
+          },
+          {
+            title: t("selectAsset.title"),
+            content: (
+              <AssetsModalContent
+                allowedAssets={allowedAssetsA}
```
Comment (vkulinich-cl) [inline src/sections/pools/modals/CreateXYKPool/CreateXYKPool.tsx:82]: does it make sense to have allowedAssets and allAssets props?

File: src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      sx={{
+        flex: "column",
+        justify: "space-between",
+        minHeight: "100%",
+      }}
+    >
+      <Controller
+        name="assetA"
+        control={form.control}
+        rules={{
```
Comment (vkulinich-cl) [inline src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.tsx:142]: why do you use rules prop if you are using zod?

File: src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        )}
+      />
+      <Separator
+        sx={{
+          mx: "calc(-1 * var(--modal-content-padding))",
+          my: 20,
+          width: "auto",
+        }}
+      />
+      <Button variant="primary" type="submit">
```
Comment (vkulinich-cl) [inline src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.tsx:205]: the Button as submit type by default

File: src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.tsx (hunk 2 above)
Comment (jvonasek) [reply src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.tsx:142]: forgot to remove, will fix

File: src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.utils.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  const { assets } = useRpcProvider()
+
+  const { isAdded } = useUserExternalTokenStore()
+
+  const accountAssets = useAcountAssets(account?.address)
+
+  return useMemo(() => {
+    const tradableAssetIds = assets.tradeAssets.map((asset) => asset.id)
+    const accountAssetIds = accountAssets
+      .filter(({ balance }) => balance.freeBalance.gt(0))
```
Comment (vkulinich-cl) [inline src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.utils.ts:61]: useAcountAssets returns only tokens with free balance, so filter fn is not necessary here

Comment (jvonasek) [reply src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.utils.ts:61]: the idea was to filter by free balance, which could be 0 if I have everything in pools or farms, if I understand correctly.

Comment (vkulinich-cl) [reply src/sections/pools/modals/CreateXYKPool/CreateXYKPoolForm.utils.ts:61]: okay, I guess it makes sense

File: src/utils/validators.ts [hunk 5]
```diff
@@ -1,13 +1,26 @@
...
 
-export const validNumber = (value: string) => {
-  if (!new BigNumber(value).isNaN()) {
-    return true
-  }
+export const required = z.string().trim().min(1, i18n.t("error.required"))
 
-  return i18n.t("error.validNumber")
-}
+export const finite = z
```
[the remark this answers is not in the document]
Comment (jvonasek) [reply src/utils/validators.ts]: updated

**Review by vkulinich-cl — APPROVED** [review]

**Pull request merged**